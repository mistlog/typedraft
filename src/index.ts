export * from "./core/transcriber";

export * from "./code-object/export-class";
export * from "./code-object/local-context";
export * from "./code-object/inline-context";
export * from "./code-object/method";
export * from "./code-object/module";

export * from "./plug-in/draft-plugin-class";
export * from "./plug-in/draft-plugin-dsl";
export * from "./plug-in/draft-plugin-filter";
export * from "./plug-in/draft-plugin-local-context";
export * from "./plug-in/draft-plugin-refresh";

export * from "./common/utility";

/**
 *
 */
import { Transcriber } from "./core/transcriber";
import { RefreshDraftPlugin } from "./plug-in/draft-plugin-refresh";
import { DSLPlugin } from "./plug-in/draft-plugin-dsl";
import { LocalContextPlugin } from "./plug-in/draft-plugin-local-context";
import { ClassPlugin } from "./plug-in/draft-plugin-class";
import { FilterPlugin } from "./plug-in/draft-plugin-filter";

export function MakeDefaultTranscriber(_module: string) {
    const transcriber = new Transcriber(_module);
    transcriber.m_Plugins = [
        new RefreshDraftPlugin(transcriber),
        new DSLPlugin(transcriber),
        new RefreshDraftPlugin(transcriber),
        new LocalContextPlugin(transcriber),
        new ClassPlugin(transcriber),
        new FilterPlugin(transcriber),
    ];
    return transcriber;
}
