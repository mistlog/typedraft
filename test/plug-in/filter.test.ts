import { MakeDefaultTranscriber, FilterPlugin, RefreshDraftPlugin } from "../../src";

test("filter class method & local context", () => {
    //
    const code = `
        export interface IFoo {
            foo: number;
        }

        export class Foo {

        }
        
        function Normal() {

        }

        // to be removed
        <Foo/> + function foo() {
            <Snippet/>;
        };

        function Snippet() {
            console.log("hi");
        }
    `;

    const transcriber = MakeDefaultTranscriber(code);
    transcriber.m_Plugins = [new RefreshDraftPlugin(transcriber), new FilterPlugin(transcriber)];

    const result = transcriber.Transcribe();
    expect(result).toMatchSnapshot();
});
