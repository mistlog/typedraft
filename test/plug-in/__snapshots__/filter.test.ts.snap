// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter class method & local context 1`] = `
"export interface IFoo {
  foo: number;
}
export class Foo {}

function Normal() {}"
`;
